Observer/Config/Save.php

```diff
@@ -65,40 +65,58 @@ public function execute(Observer $observer)
             $this->cacheTypeList->cleanType(Config::TYPE_IDENTIFIER);
             $this->appConfig->reinit();
 
-            $scope = $scopes = null;
+            $scope = $scopes = \Magento\Framework\App\Config\ScopeConfigInterface::SCOPE_TYPE_DEFAULT;
             if (($scopeId = $observer->getEvent()->getStore())) {
                 $scope = ScopeInterface::SCOPE_STORE;
                 $scopes = ScopeInterface::SCOPE_STORES;
             } elseif (($scopeId = $observer->getEvent()->getWebsite())) {
                 $scope = ScopeInterface::SCOPE_WEBSITE;
+                $scopes = ScopeInterface::SCOPE_WEBSITES;
+            } else {
+                $scopeId = 0;
             }
-            $appKey = $this->yotpoConfig->getAppKey(($scopeId ?: null), ($scope ?: null));
+
+            $appKey = $this->yotpoConfig->getAppKey($scopeId, $scope);
 
             if (in_array(YotpoConfig::XML_PATH_YOTPO_DEBUG_MODE_ENABLED, $changedPaths)) {
                 $this->yotpoConfig->log(
-                    "Yotpo Debug mode " . (($this->yotpoConfig->isDebugMode(($scopeId ?: null), ($scope ?: null))) ? 'started' : 'stopped'),
+                    "Yotpo Debug mode " . (($this->yotpoConfig->isDebugMode($scopeId, $scope)) ? 'started' : 'stopped'),
                     "info",
                     ['$app_key' => $appKey, '$scope' => ($scope ?: 'default'), '$scopeId' => $scopeId]
                 );
             }
 
-            if ($scope !== ScopeInterface::SCOPE_STORE) {
+            if ($scope !== ScopeInterface::SCOPE_STORE && !$this->yotpoConfig->isSingleStoreMode()) {
+                $this->resetStoreCredentials($scopeId, $scopes);
                 return true;
             }
+
             //Check if appKey is unique:
             if ($appKey) {
-                foreach ($this->yotpoConfig->getAllStoreIds(false) as $key => $storeId) {
-                    if ($storeId !== $scopeId && $this->yotpoConfig->getAppKey($storeId) === $appKey) {
-                        $this->yotpoConfig->resetStoreCredentials($scopeId);
+                foreach ($this->yotpoConfig->getAllStoreIds() as $key => $storeId) {
```

Comment:
why did you change this logic? it has nothing to do with single store mode - single store mode has only 1 store - in this logic looks for 2 different stores with the same app key. 

also - why did you remove the false parameter from the getAllStoresIds()?

Reply:
false is the default, so no need for it.